libopenage/versions/versions.cpp

```diff
@@ -1,36 +1,71 @@
 // Copyright 2020-2020 the openage authors. See copying.md for legal info.
 
-#include <gnu/libc-version.h>
+#include "versions.h"
+
+#ifndef __APPLE__
+  #include <gnu/libc-version.h>
+  #define TARGET_NOT_OSX
+#endif
+
 #include <SDL2/SDL.h>
 #include <sstream>
+#include <eigen3/Eigen/Dense>
+#include <harfbuzz/hb.h>
+#include <QtGlobal>
 
-#include "versions.h"
+#include "versions/compiletime.h"
+
+#include "../util/strings.h"
 
-using namespace std;
+namespace openage::versions {
 
-namespace openage
-{
+std::map<std::string, std::string> get_version_numbers() {
 
-map<string,string> get_version_numbers()
-{
-	map<string,string> version_numbers;
+	std::map<std::string, std::string> version_numbers;
 
-	//Add SDL version number
+	// Add SDL version number
 	SDL_version compiled;
 	SDL_VERSION(&compiled)
 
-	string s;
-	s.append(to_string(compiled.major));
-	s.append(".");
-	s.append(to_string(compiled.minor));
-	s.append(".");
-	s.append(to_string(compiled.patch));
+	SDL_version sdl_runtime_version;
+	SDL_GetVersion(&sdl_runtime_version);
+	version_numbers.emplace("SDL_runtime", util::sformat("%d.%d.%d",
+	                                                     sdl_runtime_version.major,
+	                                                     sdl_runtime_version.minor,
+	                                                     sdl_runtime_version.patch));
+
+	// Add Eigen version number
+	version_numbers.emplace("Eigen", util::sformat("%d.%d.%d",
+	                                                     EIGEN_WORLD_VERSION,
+	                                                     EIGEN_MAJOR_VERSION,
+	                                                     EIGEN_MINOR_VERSION));
 
-	version_numbers.insert(pair<string,string>("SDL", s));
+	// Add Harfbuzz version number
+	version_numbers.emplace("Harfbuzz", util::sformat("%d.%d.%d",
+	                                                     HB_VERSION_MAJOR,
+	                                                     HB_VERSION_MINOR,
+	                                                     HB_VERSION_MICRO));
 
-	//Add libc version number
-	version_numbers.insert(pair<string,string>("libc",gnu_get_libc_version()));
+	// Add Qt version number
+	version_numbers.emplace("Qt", QT_VERSION_STR);
+
+	// Add nyan version number
+	version_numbers.emplace("nyan", nyan_version);
+
+	#ifdef TARGET_NOT_OSX
+		// Add libc version number if not MacOSX
+		version_numbers.emplace("libc-runtime",gnu_get_libc_version());
+
+		version_numbers.emplace("libc-compile", util::sformat("%d.%d",
+	                                                     __GLIBC__,
+	                                                     __GLIBC_MINOR__));
```

Comment:
indentation alignment wrong: tabs to reach `version_numbers` and then spaces to fill up until the `sformat::(`